python/taichi/lang/kernel_impl.py

```diff
@@ -296,7 +296,11 @@ def extract_arg(arg, anno):
                                              element_dim] if layout == Layout.SOA else shape[
                                                  -element_dim:]
             return to_taichi_type(arg.dtype), len(shape), element_shape, layout
-        return type(arg).__name__,
+        if anno is int:
```

Comment:
What about using `cook_dtype(anno)` for it?

Reply:
`cook_dtype` doesn't deal with all data type now, such as `MatrixType`